ranger/gui/ui.py

```diff
@@ -474,6 +460,56 @@ def throbber(self, string='.', remove=False):
         else:
             self.titlebar.throbber = string
 
+    # Handles window renaming behaviour of the terminal multiplexers GNU Screen and Tmux
+    def handle_multiplexer(self, multiplexer=None, restore_required=False):
+        # helper function
+        def execute_rename(multiplexer):
+            try:
+                if multiplexer == 'TMUX':
+                    check_output(['tmux', 'set-window-option',
+                                  'automatic-rename', self._tmux_automatic_rename])
+                elif multiplexer == 'screen':
+                    check_output(['screen', '-X', 'title', self._screen_title])
+            except CalledProcessError:
+                pass
+
+        # Stores the screen window name before renaming it
+        # gives out a warning if $TERM is not "screen"
+
+        if multiplexer == 'screen' and not restore_required:
+            if os.environ['TERM'] != 'screen' and 'TMUX' not in os.environ:
+                self.fm.notify(
+                    '$TERM not set to "screen". Automatic window title not applied.', bad=True)
+            try:
+                self._screen_title = check_output(['screen', '-Q', 'title']).strip()
+            except CalledProcessError:
+                self._screen_title = None
+
+        # Stores the automatic-rename setting
+        # prints out a warning if the allow-rename in tmux is not set
+        if multiplexer == 'TMUX' and not restore_required:
+            if self.settings.update_tmux_title:
+                self._tmux_allow_rename = check_output(
+                    ['tmux', 'show-window-option', '-v', 'allow-rename']).strip()
+                if self._tmux_allow_rename == 'off':
+                    self.fm.notify('Warning: allow-rename not set in Tmux!', bad=True)
+                else:
+                    try:
+                        self._tmux_automatic_rename = check_output(
+                            ['tmux', 'show-window-option', '-v', 'automatic-rename']).strip()
+                        if self._tmux_automatic_rename == 'off':
+                            self._tmux_automatic_rename = 'on'
+                    except CalledProcessError:
+                        self._tmux_automatic_rename = None
+
+        if multiplexer is not None and not restore_required:
```

Comment:
Are you interpreting `restore_required` as whether to restore the window name *right now*?

If so this should be seperate functions.